app/addons/dataimporter/stores.js

```diff
@@ -0,0 +1,369 @@
+
+          this.goToErrorScreen('', [msg1, msg2]);
+        }.bind(this),
+        download: false,
+        skipEmptyLines: false,
+        chunk: undefined, //define function for streaming
+        beforeFirstChunk: undefined,
+      };
+    },
+
+    loadDataIntoDatabase: function (createNewDB, targetDB) {
+      this._loadingInDBInProgress = true;
+      this.triggerChange();
+      if (createNewDB) {
+        if (this.dataBaseIsnew(targetDB)) {
+          var msg1 = 'The database ' + targetDB + ' already exists.';
+          this.goToErrorScreen('', [msg1]);
+        }
+        this.loadIntoNewDB(targetDB);
+      } else {
+        this.loadDataIntoTarget(targetDB);
+      }
+    },
+
+    dataBaseIsnew: function (targetDB) {
+      return _.some(this._all_dbs, targetDB);
+    },
+
+    showErrorScreen: function () {
+      return this._showErrorScreen;
+    },
+
+    getErrorMsg: function () {
+      return this._errorMessageArray;
+    },
+
+    goToErrorScreen: function (resp, messageArray) {
+      this._loadingInDBInProgress = false;
+      this._showErrorScreen = true;
+      if (resp) messageArray.push(resp);
+      this._errorMessageArray.unshift(messageArray);
+      this.triggerChange();
+    },
+
+    loadIntoNewDB: function (targetDB) {
+      $.ajax({
+        url: FauxtonAPI.urls('databaseBaseURL', 'server', targetDB),
+        xhrFields: { withCredentials: true },
+        contentType: 'application/json; charset=UTF-8',
+        method: 'PUT'
+      }).then(function (resp) {
+        this.loadDataIntoTarget(targetDB);
+      }.bind(this), function (resp) {
+        this.importFailed();
+      }.bind(this));
+    },
+
+    loadDataIntoTarget: function (targetDB) {
+      var loadURL = FauxtonAPI.urls('document', 'server', targetDB, '_bulk_docs');
+      _.each(this._chunkedData, function (data, i) {
+        var payload = JSON.stringify({ 'docs': data });
+        var prettyprint = JSON.stringify({ 'docs': data }, null, 2);
+        $.ajax({
+          url: loadURL,
+          xhrFields: { withCredentials: true },
+          contentType: 'application/json; charset=UTF-8',
+          method: 'POST',
+          data: payload
+        }).then(function (resp) {
+          i++;
+          if (i === this._chunkedData.length ) {
+            //console.log("alldone");
+            this.successfulImport(targetDB);
+            this.init(true);
+          }
+        }.bind(this), function (resp) {
+          this.importFailed(resp, ['There was an error loading documents into ' + targetDB, 'Data that failed to load:' + prettyprint]);
+        }.bind(this));
+      }.bind(this));
+    },
+
+    successfulImport: function (targetDB) {
+      FauxtonAPI.navigate(FauxtonAPI.urls('allDocs', 'app', targetDB, '?include_docs=true'));
+    },
+
+    importFailed: function (resp, messageArray) {
+      this._loadingInDBInProgress = false;
+      this.goToErrorScreen(resp, messageArray);
+    },
+
+    getIsloadingInDBInProgress: function () {
```

Comment:
Bit of a mouthful this one. Should be uppercase for the `l` in loading, but maybe rename to something like `dbPopulationInProgress` or something even simpler.